File: Data Structures & Algorithms/combination-target-sum-ii/submission-1.cpp
class Solution {
public:
    void solve(int index, int target, vector<int> nums, vector<int> &temp, vector<vector<int>> &ans){
        if(target == 0){
            ans.push_back(temp);
            return;
        }
        if(target < 0) return;
        for(int i = index; i < nums.size(); i++){
            if(i != index && nums[i] == nums[i-1]) continue;
            temp.push_back(nums[i]);
            solve(i+1,target-nums[i], nums, temp, ans);
            temp.pop_back();
        }
    }
    vector<vector<int>> combinationSum2(vector<int>& nums, int target) {
        vector<int> temp;
        vector<vector<int>> ans;
        sort(nums.begin(), nums.end());
        solve(0,target, nums, temp, ans);
        return ans;
    }
};

File: Data Structures & Algorithms/sort-colors/submission-1.go
func sortColors(nums []int) {
    i, j, k := 0, 0, len(nums)-1
	for j <= k {
		if nums[j] == 0 {
			temp := nums[i]
			nums[i] = nums[j]
			nums[j] = temp
			i += 1;
			if i-1 == j {
				j += 1
			}
		} else if nums[j] == 2 {
			temp := nums[k]
			nums[k] = nums[j]
			nums[j] = temp
			k -= 1
		} else {
			j += 1
		}
	}
}
